### PR TITLE
feat: Add filter modal for mobile, move reviews component, and fix desktop filter bug on product list

### PR DESCRIPTION
**Pull request by gustavop-dev**
### Description
This pull request includes the following changes:
- **Add filter modal for the product list on mobile devices:** Implemented a modal for filtering products on mobile, enhancing the user experience on smaller screens.
- **Move reviews component in product detail view:** The reviews component has been repositioned to improve layout and readability.
- **Fix bug in desktop version filters:** Resolved an issue affecting the filters on the product list for desktop users.

### Changes
- Implemented a modal with filter options for mobile devices.
- Repositioned the reviews component in the product detail view.
- Fixed a bug causing filters to behave unexpectedly on the desktop version of the product list.

### How to Test
1. **Mobile Filter Modal:**
    - Navigate to the product list on a mobile device.
    - Open the filter modal and verify that it displays correctly and allows filtering products.

2. **Reviews Component:**
    - Go to a product detail view.
    - Ensure the reviews component is now correctly positioned and functions as expected.

3. **Desktop Filters Bug Fix:**
    - Visit the product list on a desktop device.
    - Test the filters to ensure they are working correctly without any issues.

### Additional Notes
- Make sure to test on both mobile and desktop to verify the changes.
- Check for any edge cases that might affect the filter functionality.

### CODE REVIEW

File: frontend/src/components/product/ModalFilters.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            @click="closeModal()" 
+            class="absolute inset-0 bg-gray-500 bg-opacity-40 backdrop-blur-md">
+        </div>
+        <div ref="filters" class="relative w-3/4 max-h-full h-fit px-4 mx-auto bg-white rounded-lg">
+            <div class="h-8 pt-2 flex items-center justify-end">
+                <XMarkIcon @click="closeModal()" class="w-5 h-5 text-primary_p cursor-pointer"></XMarkIcon>
+            </div>
+            <div class="overflow-auto max-h-[50vh]">
+                <CategoryFilter v-if="currentLanguage === 'en'" :categories="categories"
+                                :currentLanguage="currentLanguage"></CategoryFilter>
```
Comment (carlos18bp) [inline frontend/src/components/product/ModalFilters.vue:15]: Change to:
```
<CategoryFilter v-if="currentLanguage === 'en'" 
   :categories="categories"
   :currentLanguage="currentLanguage">
</CategoryFilter>
```

File: frontend/src/views/product/Detail.vue [hunk 2]
```diff
@@ -289,6 +265,31 @@
...
+            {{ $t('carousel_subtitle') }}
+        </h2>
+        <ProductCarousel :top="4"></ProductCarousel>
+    </div>
+
+    <!-- Trending Section -->
+    <div class="container mx-auto mt-16 grid text-center xl:text-start xl:grid-cols-2 gap-4">
+        <div class="grid content-center">
+            <div>
+                <h2 class="font-regular text-2xl text-gray_p tracking-widest">{{ $t('trending') }}</h2>
```
Comment (carlos18bp) [inline frontend/src/views/product/Detail.vue:284]: Change to:
```
<h2 class="font-regular text-2xl text-gray_p tracking-widest">
   {{ $t('trending') }}
</h2>
```

File: frontend/src/views/product/Detail.vue [hunk 3]
```diff
@@ -289,6 +265,31 @@
...
+                <p class="font-medium text-2xl xl:text-4xl text-black_p mt-12 tracking-wider">{{ $t('text') }}</p>
```
Comment (carlos18bp) [inline frontend/src/views/product/Detail.vue:285]: Change to:
```
<p class="font-medium text-2xl xl:text-4xl text-black_p mt-12 tracking-wider">
   {{ $t('text') }}
</p>
```

File: frontend/src/views/product/List.vue [hunk 4]
```diff
@@ -5,10 +5,16 @@
...
             <main class="mx-auto xl:max-w-7xl">
                 <section aria-labelledby="products-heading">
-                    <div class="grid gap-x-8 gap-y-10 grid-cols-4">
-                        
-                        <!-- Filters -->
-                        <div class="col-span-4 md:col-span-1">
+                    <div class="grid gap-x-8 grid-cols-4 lg:gap-y-10">
+                        <!-- Filters for screens shorter than lg -->
+                        <div class="block col-span-4 lg:hidden">
+                            <h1 @click="showModalFilters = true" class="flex items-center gap-2 text-xl font-bold tracking-tight text-gray-900 cursor-pointer test-filter_title">
```
Comment (carlos18bp) [inline frontend/src/views/product/List.vue:11]: Change to:
```
 <h1 @click="showModalFilters = true" 
   class="flex items-center gap-2 text-xl font-bold tracking-tight text-gray-900 cursor-pointer test-filter_title">
```

**Review by carlos18bp — CHANGES_REQUESTED** [review]

File: frontend/src/views/product/List.vue (hunk 4 above)
Comment (gustavop-dev) [reply frontend/src/views/product/List.vue:11]: Done!, You can preview the changes in candle-g-21-1

File: frontend/src/views/product/Detail.vue (hunk 3 above)
Comment (gustavop-dev) [reply frontend/src/views/product/Detail.vue:285]: Done!, You can preview the changes in candle-g-21-1

File: frontend/src/views/product/Detail.vue (hunk 2 above)
Comment (gustavop-dev) [reply frontend/src/views/product/Detail.vue:284]: Done!, You can preview the changes in candle-g-21-1

File: frontend/src/components/product/ModalFilters.vue (hunk 1 above)
Comment (gustavop-dev) [reply frontend/src/components/product/ModalFilters.vue:15]: Done!, You can preview the changes in candle-g-21-1

File: frontend/src/components/product/CategoryFilter.vue [hunk 5]
```diff
@@ -12,12 +12,13 @@
...
         </h3>
-
-        <!-- Sub-category filter for English language -->
-        <SubCategoryFilter v-if="currentLanguage === 'en'" :subCategories="category.subCategories" />
-        
-        <!-- Sub-category filter for other languages -->
-        <SubCategoryFilter v-else :subCategories="category.subCategorias" />
+        <div v-if="open" class="pt-6">
+            <!-- Sub-category filter for English language -->
+            <SubCategoryFilter v-if="currentLanguage === 'en'" :subCategories="category.subCategories" />
```
Comment (carlos18bp) [inline frontend/src/components/product/CategoryFilter.vue:17]: Change to:
```
<SubCategoryFilter v-if="currentLanguage === 'en'" 
   :subCategories="category.subCategories" />
```

File: frontend/src/components/product/CategoryFilter.vue [hunk 6]
```diff
@@ -12,12 +12,13 @@
...
+            
+            <!-- Sub-category filter for other languages -->
+            <SubCategoryFilter v-if="currentLanguage === 'es'" :subCategories="category.subCategorias" />
```
Comment (carlos18bp) [inline frontend/src/components/product/CategoryFilter.vue:20]: Change to:
```
<SubCategoryFilter v-if="currentLanguage === 'es'" 
   :subCategories="category.subCategorias" />
```

File: frontend/src/views/product/Checkout.vue [hunk 7]
```diff
@@ -109,12 +81,13 @@
...
-                            class="w-36 bg-terciary_p text-white p-3 rounded-lg hover:bg-primary_p font-semibold text-lg tracking-wider test-pay_now">
-                            {{ $t("pay_now") }}
-                        </button>
+                    <h2 class="text-3xl font-semibold mt-6 test-payment_details">
+                        {{ $t("payment_details") }}
+                    </h2>
+                    <div class="mt-8">
+                        <PaymentForm ref="paymentFormRef"></PaymentForm>
                     </div>
+                    <button class="px-4 py-2 bg-primary_p rounded-lg text-white text-xl font-regular">{{ $t('pay_now') }}</button>
```
Comment (carlos18bp) [inline frontend/src/views/product/Checkout.vue:90]: Change to:
```
<button class="px-4 py-2 bg-primary_p rounded-lg text-white text-xl font-regular">
   {{ $t('pay_now') }}
</button>
```

**Review by carlos18bp — APPROVED** [review]

File: frontend/src/components/product/CategoryFilter.vue (hunk 5 above)
Comment (gustavop-dev) [reply frontend/src/components/product/CategoryFilter.vue:17]: Done!

File: frontend/src/components/product/CategoryFilter.vue (hunk 6 above)
Comment (gustavop-dev) [reply frontend/src/components/product/CategoryFilter.vue:20]: Done!

File: frontend/src/views/product/Checkout.vue (hunk 7 above)
Comment (gustavop-dev) [reply frontend/src/views/product/Checkout.vue:90]: Done!
